Files/ViewModels/SettingsViewModels/AppearanceViewModel.cs

```diff
@@ -107,16 +108,35 @@ public AppTheme SelectedTheme
             {
                 if (SetProperty(ref selectedTheme, value))
                 {
+                    // Remove the old resource file and load the new file
+                    App.ExternalResourcesHelper.UpdateTheme(App.AppSettings.SelectedTheme, selectedTheme);
+
                     App.AppSettings.SelectedTheme = selectedTheme;
-                    ShowRestartControl = true;
+
+                    // Force the application to use the correct resource file
+                    UpdateTheme();
                 }
             }
         }
 
-        public bool ShowRestartControl
+        /// <summary>
+        /// Forces the application to use the correct resource styles
+        /// </summary>
+        private async void UpdateTheme()
         {
-            get => showRestartControl;
-            set => SetProperty(ref showRestartControl, value);
+            // Allow time to remove the old theme
+            await Task.Delay(250);
```

Comment:
Is there any way we can verify the old theme is removed without hardcoding in a delay?